wordpress-mobile/WordPress-Android
# My Site Dashboard: QuickStartCardSource + QuickStartRepository

Parent #15215

This PR continues with My Site source refactoring by moving logic out `QuickStartRepository` and into `QuickStartCardSource`.  The contents of `buildSource` have been moved to QuickStartCardSource. A few function changes were made to simplify the transition. 

Tests affected by this refactor were moved from `QuickStartRepositoryTest` was `QuickStartCardSourceTest`. In addition, `MySiteViewModelTest` and `WPMainViewModelTest` were updated to call the updated methods.

This PR is part of a series of PRs being written for adding pull-to-refresh to the MySite tab.

Please lmk if I have missed any testing scenarios below. Thanks.

@startuptester and @designsimply. Although I don't anticipate any breaking changes from this refactor, I'd feel much better knowing that a second (unbiased) set of eyes took a look. If either of you have the time, I'd appreciate you taking the Quick Start flow out for a test drive, just to make sure it still works as expected. If you do have the time, add yourself as a reviewer so the PR doesn't get merged without your approval. 🙇  🙏  Thank you!

**To test:**
Prerequisite: 

1. Go to `My Site` -> `App Settings` -> `Test feature configuration`
2. Disable `QuickStartDynamicCardsFeatureConfig` if checked & restart the app

##### Test 1 - Quick start notice display 

1. Select a site with quick start in progress 
2. Click quick start task type list item
3. Notice that `QuickStartFullScreenDialogFragment` gets opened
4. Select a task from the `QuickStartFullScreenDialogFragment`
5. Notice that when `QuickStartFullScreenDialogFragment` is dismissed, the appropriate QS focus point and snack bar get displayed based on the chosen task
6. Perform the task so that it gets completed
7. Notice that quick start notice is displayed for the next task 

##### Test 2 - snackbar action 
1. Click "Go" button in the quick start notice
2. Notice that appropriate QS focus point and snack bar get displayed based for this next task

##### Test 3   - Notifications/ Reminders 
1. Create a new site or do a fresh login
2. When the Quick Start Prompt is shown, tap the "Show me around" option
3. Tap the Customize your site row to launch the QS dialog
4. Tap on Check your site title (the dialog will close and you'll be brought back to My Site)
5. Tap on the blue flashing icon on site title and change the text so that the task gets completed
6. Navigate out of the app to the Settings area of your device and locate the area that manages the time.
7. Disable the "use network-provided time"
8. Manually set the date/time for 2 days and 2 minutes from now (as the notifications are set for 2 day in future)
9. Wait for a couple of minutes and note that the notification does arrives at the device
10. Once testing is complete, enable "use network-provided time"

##### Test 4   - Notifications/ Reminders after Quick Start or "Next Steps" section is removed
1. Create a new site or do a fresh login
2. When the Quick Start Prompt is shown, tap the "Show me around" option
3. Tap the Customize your site row to launch the QS dialog
4. Tap on Check your site title (the dialog will close and you'll be brought back to My Site)
5. Tap on the blue flashing icon on site title and change the text so that the task gets completed
6. Click more menu -> "Remove" menu item -> "Remove" button in the "Remove Next Steps" dialog so that Quick Start or" Next Steps" section is removed
7. Navigate out of the app to the Settings area of your device and locate the area that manages the time.
8. Disable the "use network-provided time"
9. Manually set the date/time for 2 days and 2 minutes from now (as the notifications are set for 2 day in future)
10. Wait for a couple of minutes and note that the notification does NOT arrive at the device
11. Once testing is complete, enable "use network-provided time"

##### Test 5  - Quick Start Cards disappear
1. Create a new site or do a fresh login
2. When the Quick Start Prompt is shown, tap the "Show me around" option
3. Tap the Customize your site row to launch the QS dialog
4. Follow all the prompts and complete all customize tasks
5. Tap the Grow Your Audience row to launch the QS dialog
6. Follow all the prompts and complete all the growth tasks
7. When all steps are complete, note the Quick Start card has been removed from the My Site view
- 
## Regression Notes
1. Potential unintended areas of impact
Quick Start actions/tasks don't work as expected

2. What I did to test those areas of impact (or what existing automated tests I relied on)
- Manual testing, executed automated tests

3. What automated tests I added (or what prevented me from doing so)
- Updated unit tests to reflect refactored code

PR submission checklist:


FYI: @ParaskP7
